lib/sqlalchemy/dialects/postgresql/asyncpg.py

```diff
@@ -915,10 +915,9 @@ def terminate(self):
             try:
                 # try to gracefully close; see #10717
                 # timeout added in asyncpg 0.14.0 December 2017
-                await_(self._connection.close(timeout=2))
+                await_(asyncio.shield(self._connection.close(timeout=2)))
             except (
                 asyncio.TimeoutError,
-                asyncio.CancelledError,
```

Comment:
why remove CancelledError here?  is it 100% impossible for CancelledError to be raised from sheild()?

Reply:
re-added